store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -2743,6 +2850,258 @@ public String getServiceName() {
 
     }
 
+    class MainBatchDispatchRequestService extends ServiceThread {
+
+        private final ExecutorService batchDispatchRequestExecutor;
+
+        public MainBatchDispatchRequestService() {
+            batchDispatchRequestExecutor = new ThreadPoolExecutor(
+                    DefaultMessageStore.this.getBrokerConfig().getBatchDispatchRequestThreadPoolNums(),
+                    DefaultMessageStore.this.getBrokerConfig().getBatchDispatchRequestThreadPoolNums(),
+                    1000 * 60,
+                    TimeUnit.MICROSECONDS,
+                    new LinkedBlockingQueue<>(1024),
```

Comment:
when runnable tasks greater than  1024, batchDispatchRequestExecutor miss RejectedExecutionHandler  logic，the default RejectedExecutionHandler is right?

Reply:
> when runnable tasks greater than 1024, batchDispatchRequestExecutor miss RejectedExecutionHandler logic，the default RejectedExecutionHandler is right?

The reason I set the value to 1024 is that I think this size is sufficient. Because a task size is 4MB. A commitlog file size of 1G can generate 256tasks. So the queue can hold 1024 tasks generated by four commitlog files. So I set the default reject policy.

Reply:
Does it need a policy to temporarily save more than 1024 tasks? I can adjust it if necessary.

Reply:
> > when runnable tasks greater than 1024, batchDispatchRequestExecutor miss RejectedExecutionHandler logic，the default RejectedExecutionHandler is right?
> 
> The reason I set the value to 1024 is that I think this size is sufficient. Because a task size is 4MB. A commitlog file size of 1G can generate 256tasks. So the queue can hold 1024 tasks generated by four commitlog files. So I set the default reject policy.

I got it .

adjust 4096 ?

Reply:
> 

okk.

Reply:
@nowinkeyy , sorry.

I thought for a moment，it must be se right RejectedExecutionHandler,because if it occur,may can lost message. the 
reject policy can block task,but cant't throw Exception cause message lost

Reply:
> @nowinkeyy , sorry.
> 
> I thought for a moment，it must be se right RejectedExecutionHandler,because if it occur,may can lost message. the reject policy can block task,but cant't throw Exception cause message lost

You are right. And i have an idea that I don't know if it is feasible. Before putting a task on a thread pool queue, determine if the queue size is less than 1024? As shown in the figure:
![image](https://user-images.githubusercontent.com/72536832/216774028-71396e47-3c39-452e-b7ea-913145f2f2c3.png)

Reply:
i can't confirm that batchDispatchRequestQueue queue getSize method is concurrent safe

common usage  set right  reject policy of Executor.

Reply:
> i can't confirm that batchDispatchRequestQueue queue getSize method is concurrent safe
> 
> common usage set right reject policy of Executor.

You are right, it is more appropriate to set a right reject policy of Executor. 

( The getsize() method is concurrency-safe because it is LinkedBlockingQueue. )